### ISSUE TITLE
staticcheck: panic in CheckSillyBitwiseOps

### ISSUE THREAD

**Issue by JasonTashtego** [dominikh/go-tools#838]
Please make sure to include the following information in your issue report:

- The output of 'staticcheck -version'

```
staticcheck 2020.1.5
```

- The output of 'staticcheck -debug.version' (it is fine if this command fails)

```
staticcheck 2020.1.5

Compiled with Go version: go1.14.4
Main module:
        honnef.co/go/tools@v0.0.1-2020.1.5 (sum: h1:nI5egYTGJakVyOryqLs1cQO5dO0ksin5XXs2pspk75k=)
Dependencies:
        github.com/BurntSushi/toml@v0.3.1 (sum: h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=)
        golang.org/x/tools@v0.0.0-20191130070609-6e064ea0cf2d (sum: h1:/iIZNFGxc/a7C3yWjGcnboV+Tkc7mxr+p6fDztwoxuM=)
```

- The output of 'go version'

```
go version go1.14.9 windows/amd64
```

- The output of 'go env'

```
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\jason\AppData\Local\go-build
set GOENV=C:\Users\jason\AppData\Roaming\go\env
set GOEXE=
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GONOPROXY=
set GONOSUMDB=
set GOOS=linux
set GOPATH=D:\treat\go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=c:\go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=0
set GOMOD=D:\txxt\src\sttxex\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-fPIC -m64 -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=C:\Users\jason\AppData\Local\Temp\go-build524836414=/tmp/go-build -gno-record-gcc-switches
```

- Exactly which command you ran

```
staticcheck -checks "all"
```

- Output of the command and what's wrong with the output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x51ced9]

goroutine 147327 [running]:
go/ast.(*File).End(0x0, 0x1c00)
        go/ast/ast.go:1000 +0x29
golang.org/x/tools/go/ast/astutil.PathEnclosingInterval(0x0, 0x84105ed, 0x84105ed, 0x0, 0x0, 0x0, 0x2b4c455a)
        golang.org/x/tools@v0.0.0-20191130070609-6e064ea0cf2d/go/ast/astutil/enclosing.go:140 +0xd8
honnef.co/go/tools/staticcheck.CheckSillyBitwiseOps.func1(0xb97c40, 0xc013c145a0)
        honnef.co/go/tools@v0.0.1-2020.1.5/staticcheck/lint.go:2717 +0x1e2
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc03880ff40, 0xc002087bd8, 0x1, 0x1, 0xc002087be8)
        golang.org/x/tools@v0.0.0-20191130070609-6e064ea0cf2d/go/ast/inspector/inspector.go:77 +0xa6
honnef.co/go/tools/code.Preorder(...)
        honnef.co/go/tools@v0.0.1-2020.1.5/code/code.go:480
honnef.co/go/tools/staticcheck.CheckSillyBitwiseOps(0xc004a005a0, 0x1c2838e6b1, 0xf87c80, 0xc021842948, 0x0)
        honnef.co/go/tools@v0.0.1-2020.1.5/staticcheck/lint.go:2758 +0xcc
honnef.co/go/tools/lint.(*Runner).runAnalysisUser(0xc0002c5f80, 0xc004a005a0, 0xc036c360f0, 0x0, 0x0, 0x0, 0x0)
        honnef.co/go/tools@v0.0.1-2020.1.5/lint/runner.go:494 +0x2b6
honnef.co/go/tools/lint.(*Runner).runAnalysis(0xc0002c5f80, 0xc036c360f0, 0x0, 0x0, 0x0, 0x0)
        honnef.co/go/tools@v0.0.1-2020.1.5/lint/runner.go:374 +0x57b
honnef.co/go/tools/lint.(*Runner).processPkg.func3(0xc001093a50, 0xc001328360, 0xc0002de270, 0xc0002c5f80, 0xc036c360f0, 0xc037f32000, 0x88, 0x88, 0x4a)
        honnef.co/go/tools@v0.0.1-2020.1.5/lint/runner.go:913 +0x80
created by honnef.co/go/tools/lint.(*Runner).processPkg
        honnef.co/go/tools@v0.0.1-2020.1.5/lint/runner.go:908 +0x9fa
```

- Where we can read the code you're running staticcheck on
  (GitHub repo, link to playground, code embedded in the issue, ...)

(private)

**Comment by dominikh** [comment]
I will need example code that triggers the crash to be able to debug it.

**Comment by JasonTashtego** [comment]
Is there a check I can disable to avoid CheckSillyBitwiseOps and still produce the rest of the output ? 

Due to restrictions I cannot supply the code it as it exists.  I will work on producing a narrower case and file new when I can produce it.

**Comment by dominikh** [comment]
> Is there a check I can disable to avoid CheckSillyBitwiseOps and still produce the rest of the output ?

There is not. Disabled checks still run (so that their results can be cached), and thus would still run into the crash.

However, this bug is probably a duplicate of https://github.com/dominikh/go-tools/issues/719, which we seem to have neglected to backport to Staticcheck 2020.1.5. Will fix.

**Comment by JasonTashtego** [comment]
Ok, thanks. I'll retest it then

**Comment by dominikh** [comment]
This has been fixed in Staticcheck 2020.1.6.
